### PR TITLE
Enabled diag for bool tensor

### PR DESCRIPTION
**Pull request by izdeby**
Fixed this [issue](https://github.com/pytorch/pytorch/issues/30923)

### CODE REVIEW

**Comment by kostmo** [comment]
## CircleCI build failures summary
As of commit b790ca8:
* **1/1** broken upstream at merge base 3a02ed8 (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=3a02ed822b13bf914f8f84fda241edb950c9312f&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
* **0/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=b790ca8c2eb9d64e6b52cd002d2bef56b715725d).

### 1 failure *not* recognized by patterns:
| Job | Step |
| --- | --- |
| CircleCI <sup>pytorch_xla_linux_xenial_py3_6_clang7_build</sup> | <sup>Build</sup> |


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

File: test/test_torch.py [hunk 1]
```diff
@@ -6405,6 +6405,13 @@ def test_diagonal(self, device):
         expected = torch.diag(x, 17)
         self.assertEqual(result, expected)
 
+        # test bool tensor
+        a = torch.tensor([True, False, True], device=device)
+        expected = torch.tensor([[True, False, False],
+                                 [False, False, False],
+                                 [False, False, True]])
+        self.assertEqual(a.diag(), expected)
```
Comment (vishwakftw) [inline test/test_torch.py]: Shouldn't this be in `test_diag` and not in `test_diagonal`?

Comment (ezyang) [reply test/test_torch.py]: Also, it seems to me like this test should be dtype-generic.

Comment (izdeby) [reply test/test_torch.py]: Added additional test as rand in not supported for bool tensor.

**Review by vishwakftw — APPROVED** [review]
Looks good to me. But as @ezyang said couldn't we add this as a dtype generic test?

Like:
```
a = torch.randint(0, 2, size=(size,)).to(dtype=dtype)
# Now perform tests
```

**Pull request closed without merging**